### PR TITLE
Modify S5886: LaYC format

### PR DESCRIPTION
**Pull request by guillaume-dequenne-sonarsource**
## Review

A dedicated reviewer checked the rule description successfully for:

- [ ] logical errors and incorrect information
- [ ] information gaps and missing content
- [ ] text style and tone
- [ ] PR summary and labels follow [the guidelines](https://github.com/SonarSource/rspec/#to-modify-an-existing-rule)

### CODE REVIEW

File: rules/S5886/python/rule.adoc [hunk 1]
```diff
@@ -26,7 +51,10 @@ def hello() -> str:
 
 == Resources
 
-* https://docs.python.org/3/library/typing.html[Python documentation - Support for type hints]
+=== Documentation
+
+* Python documentation https://docs.python.org/3/library/typing.html[Support for type hints]
```
Comment (joke1196) [inline rules/S5886/python/rule.adoc:56]: Small thing but a `-` is missing between the name and the link

File: rules/S5886/python/rule.adoc [hunk 2]
```diff
@@ -1,23 +1,48 @@
...
+def collect() -> Iterable: return Bucket()
+----
+
+Since type annotations are not enforced at runtime, returning a completely different type might not fail. It is however likely to be unintended and will lead to maintainability issues, if not bugs.
+
+== How to fix it
+
+=== Code examples
+
+To fix this issue, make sure that the returned values of your function are compatible with its type hint.
```
Comment (joke1196) [inline rules/S5886/python/rule.adoc:32]: Feel free to ignore this one, but I would rather go with `returned value ... is compatible`. I feel it is weird that a function could return multiple types.

**Review by joke1196 — CHANGES_REQUESTED** [review]
LGTM. I have just added small comments nothing major so feel free to ignore them. I really like this rule's description it is super clear!

File: rules/S5886/python/rule.adoc [hunk 3]
```diff
@@ -26,7 +51,10 @@ def hello() -> str:
 
 == Resources
 
-* https://docs.python.org/3/library/typing.html[Python documentation - Support for type hints]
+=== Documentation
+
+* Python documentation - https://docs.python.org/3/library/typing.html[Support for type hints]
+* PEP 544 https://peps.python.org/pep-0544/[Protocols: Structural subtyping (static duck typing)]
```
Comment (joke1196) [inline rules/S5886/python/rule.adoc:57]: Sorry to be super annoying here but a `-` is still missing

**Review by joke1196 — APPROVED** [review]
LGTM. I still added a micro comment for a missing `-`.

File: rules/S5886/python/rule.adoc (hunk 3 above)
Comment (guillaume-dequenne-sonarsource) [reply rules/S5886/python/rule.adoc]: No worries, that's my bad! Thanks!

File: rules/S5886/python/rule.adoc [hunk 4]
```diff
@@ -1,23 +1,48 @@
...
 
-
-=== Noncompliant code example
+In the following example, while `Bucket` does not directly inherit from `Iterable`, it does implement the `Iterable` protocol thanks to its ``++__iter__++`` method and can therefore be used as a valid `Iterable` return type.
 
 [source,python]
 ----
+from collections.abc import Iterator, Iterable
+
+class Bucket:  # Note: no base classes
```
Comment (maksim-grebeniuk-sonarsource) [inline rules/S5886/python/rule.adoc:17]: maybe it is better to move it to compliant/noncompliant samples? Showing how it could looks like in correct way

**Review by maksim-grebeniuk-sonarsource — APPROVED** [review]
LGTM, just left a minor suggestion. Feel free to ignore it.

**Review by jean-jimbo-sonarsource — APPROVED** [review]

File: rules/S5886/python/rule.adoc (hunk 4 above)
Comment (guillaume-dequenne-sonarsource) [reply rules/S5886/python/rule.adoc:17]: The intention was to keep the compliant/noncompliant examples as simple as possible and to have this slightly more complex case as a standalone example.

**Pull request merged**